File: LeetCode-September-Challenge/BinaryTreeInorderTraversal.js
function TreeNode(val, left, right) {
    this.val = (val===undefined ? 0 : val)
    this.left = (left===undefined ? null : left)
    this.right = (right===undefined ? null : right)
}
var inorderTraversal = function (root) {
    let res = new Array();
    const inorder = (root) => {
        if(root){
            inorder(root.left);
            res.push(root.val);
            inorder(root.right);
        }
    }
    inorder(root);
    return res;
};

File: Year-2022/LeetCode-September-Challenge/BinaryTreeInorderTraversal.java
import java.util.*;
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}
class Solution {
    private List<Integer> res = new ArrayList<>();
    public List<Integer> inorderTraversal(TreeNode root) {
        if(root != null){
            res = inorderTraversal(root.left);
            res.add(root.val);
            res = inorderTraversal(root.right);
        }
        return res;
    }
}
public class BinaryTreeInorderTraversal{
    public static void main(String[] args) {
        
    }
}